test/fuzz/equivalence_relation_test.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <set>
+
+#include "gtest/gtest.h"
+#include "source/fuzz/equivalence_relation.h"
+
+namespace spvtools {
+namespace fuzz {
+namespace {
+
+struct UInt32Equals {
+  bool operator()(const uint32_t* first, const uint32_t* second) const {
+    return *first == *second;
+  }
+};
+
+struct UInt32Hash {
+  size_t operator()(const uint32_t* element) const {
+    return static_cast<size_t>(*element);
+  }
+};
+
+std::set<uint32_t> ToUIntSet(
+    EquivalenceRelation<uint32_t, UInt32Hash, UInt32Equals>::ValueSet
+        pointers) {
+  std::set<uint32_t> result;
+  for (auto pointer : pointers) {
```

Comment:
Traversing any set or unordered set where the key is a pointer is non-deterministic.  It does not seem to be an issue in this case.

Reply:
In #3006 I have changed the interface to EquivalenceRelation so that it does not expose unordered sets.